docs/Compiling Non-BEAM Code.md

```diff
@@ -0,0 +1,170 @@
+# Compiling Non-BEAM Code
+
+It's almost guaranteed that you'll have some code in your project that won't be
+written in Elixir, Erlang, or another BEAM language. Nerves provides multiple
+ways of integrating this code and the one you choose depends on many things.
+
+Here are rules of thumb:
+
+1. Build large and complicated C and C++ projects using Buildroot by creating a
+   [Custom system](https://hexdocs.pm/nerves/customizing-systems.html)
+2. Build small C and C++ projects using
+   [`elixir_make`](https://hex.pm/packages/elixir_make)
+3. Look for libraries like [`zigler`](https://hex.pm/packages/zigler) for
+   specific languages
+4. When hope is lost, compile the programs outside of Nerves and include the
+   binaries in a `priv` directory. Static linking is recommended.
+
+In a perfect world, it would be easy to use whatever language you wanted and
+adding a program would be as simple as adding a reference to it to your `mix
+deps`. Sadly, that's not the case for embedded systems and sometimes an inferior
+library may be preferable just because it carries fewer dependencies or is
+easier to build.
+
+Be aware of the following caveats with Nerves:
+
+1. Nerves does not use the embedded Linux init systems like `systemd` or
+   `BusyBox init`. Initialization is done in either an
+   [Application.start callback](https://hexdocs.pm/elixir/Application.html#module-the-application-callback-module)
+   or in a `GenServer` so that it can be supervised.
+2. D-Bus is not normally enabled on Nerves. It may be enabled in a custom
+   system.
+3. X Windows is not used. Again, it may be enabled, but it is far more common to
+   have UI applications be fullscreen and not use a window manager.
+4. Minimal commands are available for shell scripts. Nerves ships with the
+   [`ash`](https://en.wikipedia.org/wiki/Almquist_shell) and bash scripts
+   nearly always require adjustments.
```

Comment:
Im not quite sure what is trying to be said here? Nerves has `ash`, but bash requires adjustments to be used?